File: src/main/java/binarysearch/InorderTraversal.java
package binarysearch;

import java.util.ArrayList;
import java.util.List;

public class InorderTraversal {
    private List<Integer> ret = new ArrayList<>();

    public int[] solve(Tree root) {
        go(root);
        int[] rett = new int[ret.size()];
        for (int i = 0; i < ret.size(); i++) rett[i] = ret.get(i);
        return rett;
    }

    private void go(Tree root) {
        if (root == null) return;
        go(root.left);
        ret.add(root.val);
        go(root.right);
    }

    public static class Tree {
        public int val;
        public Tree left;
        public Tree right;
    }
}

File: src/main/cpp/binarysearch/inorder_traversal.cpp
#include <bits/stdc++.h>

using namespace std;

struct Tree {
	int val;
	Tree *left, *right;
};

void go(Tree* root, vector<int>& acc) {
    if (!root) return;
    go(root->left, acc);
    acc.push_back(root->val);
    go(root->right, acc);
}

vector<int> solve(Tree* root) {
    vector<int> ans;
    go(root, ans);
    return ans;
}